src/compiler/utilities.ts

```diff
@@ -3294,11 +3294,21 @@ namespace ts {
         return !!name && isDynamicName(name);
     }
 
+    export function getComputedPropertyNameExpression(name: ComputedPropertyName): Expression {
+        return isElementAccessExpression(name.expression)
+                    ? skipParentheses(name.expression.argumentExpression)
+                    : name.expression;
+    }
+
     export function isDynamicName(name: DeclarationName): boolean {
         if (!(name.kind === SyntaxKind.ComputedPropertyName || name.kind === SyntaxKind.ElementAccessExpression)) {
             return false;
         }
-        const expr = isElementAccessExpression(name) ? skipParentheses(name.argumentExpression) : name.expression;
+
+        const expr = isElementAccessExpression(name)
+        ? skipParentheses(name.argumentExpression)
+        : getComputedPropertyNameExpression(name);
+

```

Comment:
in the case of a `ComputedPropertyName` with `ElementAccessExpression` -
`expr` ended up being `ElementAccessExpression` instead of some literal 

`getComputedPropertyNameExpression` was added to extract it